app/lib/shared/markdown.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:markdown/markdown.dart' as m;
+import 'package:path/path.dart' as p;
+
+final List<m.BlockSyntax> _blockSyntaxes =
+    m.ExtensionSet.gitHub.blockSyntaxes.toList();
+
+final List<m.InlineSyntax> _inlineSyntaxes = m
+    .ExtensionSet.gitHub.inlineSyntaxes
+    .where((s) => s is! m.InlineHtmlSyntax)
+    .toList();
+
+String markdownToHtml(String text, String baseUrl) {
+  return m.markdownToHtml(
+    text,
+    extensionSet: _createCustomExtension(baseUrl),
+    blockSyntaxes: _blockSyntaxes,
+    inlineSyntaxes: _inlineSyntaxes,
+  );
+}
+
+class _RelativeLinkSyntax extends m.LinkSyntax {
+  final Uri baseUri;
+  _RelativeLinkSyntax(this.baseUri);
+
+  static _RelativeLinkSyntax parse(String url) {
+    try {
+      final Uri uri = Uri.parse(url);
+      if (uri.scheme != 'http' && uri.scheme != 'https') return null;
+      if (uri.host == null || uri.host.isEmpty || !uri.host.contains('.'))
+        return null;
```

Comment:
For multiline if's we use {}s. Maybe there's an analyzer linter flag we can turn on?